### PR TITLE
Introduce a ViewportBlueprint archetype and use it for querying the Viewport

### PR DESCRIPTION
**Pull request by jleibs**
### What
Rather than querying a bunch of components in an adhoc way, we now do a single query using a new archetype.

There's still a fair bit of manual type-conversion and wrapping to be cleaned up in the future, but this is a big step
toward the structure that we want.

Best reviewed commit-by-commit.

TODO:
 - [x] Fix cpp tests

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Full build: [app.rerun.io](https://app.rerun.io/pr/4504/index.html)
  * Partial build: [app.rerun.io](https://app.rerun.io/pr/4504/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json) - Useful for quick testing when changes do not affect examples in any way
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4504)
- [Docs preview](https://rerun.io/preview/84105c955d339a20426cd54c3606a9b334a22c16/docs) 
- [Examples preview](https://rerun.io/preview/84105c955d339a20426cd54c3606a9b334a22c16/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_types/definitions/rerun/blueprint/components/included_space_views.fbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/// The id of a `SpaceView`.
+///
+/// Unstable. Used for the ongoing blueprint experimentations.
+table IncludedSpaceViews (
+  "attr.rerun.scope": "blueprint",
+  "attr.rust.derive": "Default",
+  "attr.rust.override_crate": "re_viewport",
+  "attr.rust.repr": "transparent"
+) {
+  space_view: [rerun.datatypes.Uuid] (order: 100);
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/blueprint/components/included_space_views.fbs:22]: shouldnt this be `space_views` since it's several?

**Review by Wumpf — APPROVED** [review]
amazing!
You made this look so smooth 😄

File: crates/re_query/src/archetype_view.rs [hunk 2]
```diff
@@ -486,6 +486,15 @@ impl<A: Archetype> ArchetypeView<A> {
             phantom: PhantomData,
         }
     }
+
+    /// Convert an `ArchetypeView` back into a native Archetype instance
+    pub fn as_native(&self) -> crate::Result<A> {
```
[the remark this answers is not in the document]
Comment (emilk) [reply crates/re_query/src/archetype_view.rs:491]: This should be called `to_native`: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

File: crates/re_viewport/src/blueprint/components/auto_layout_ext.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+use super::AutoLayout;
+
+impl Default for AutoLayout {
+    fn default() -> Self {
```
Comment (emilk) [inline crates/re_viewport/src/blueprint/components/auto_layout_ext.rs:4]:
```suggestion
    #[inline]
    fn default() -> Self {
```

File: crates/re_query/src/archetype_view.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wumpf) [reply crates/re_query/src/archetype_view.rs:491]: `to_archetype` then 😄

**Pull request merged**